### PR TITLE
Refactor global config imports 

### PR DESCRIPTION
**Pull request by Yun-Kim**
## Description
Currently, the global `config` object is instantiated in `ddtrace.settings`, then imported by the tracer and library patches from settings. It makes more sense and is less messy to change the library patch imports to be from `ddtrace` rather than `settings`. Additionally, the use of mypy type checks results in mypy mistaking the `config` import to be a module import rather than an attribute, which is further motivation for this refactor. 

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Review by Kyle-Verhoog — COMMENTED** [review]
I don't see the actual change for moving `ddtrace.settings.config` to `ddtrace`, was that meant to be included as well?

**Comment by Yun-Kim** [comment]
> I don't see the actual change for moving `ddtrace.settings.config` to `ddtrace`, was that meant to be included as well?

I checked and the only time `settings.config` was referenced was for the modifications I updated here, will `settings.config` be used again?

File: ddtrace/settings/__init__.py [hunk 1]
```diff
@@ -5,9 +5,6 @@
 from .integration import IntegrationConfig
 
 
-# Default global config
-config = Config()
-
 __all__ = [
     "config",
```
Comment (Kyle-Verhoog) [inline ddtrace/settings/__init__.py:9]:
```suggestion
```

File: ddtrace/settings/__init__.py [hunk 2]
```diff
@@ -5,9 +5,6 @@
 from .integration import IntegrationConfig
 
 
-# Default global config
-config = Config()
```
Comment (Kyle-Verhoog) [inline ddtrace/settings/__init__.py:9]: This should be moved to `ddtrace/__init__.py`

Comment (Kyle-Verhoog) [reply ddtrace/settings/__init__.py:9]: Ended up not being possible as there's a circular dependency between `ddtrace` and `ddtrace.monkey`

**Review by Kyle-Verhoog — APPROVED** [review]
LGTM - might want to update the PR title though 🙂

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**